### PR TITLE
nixos/restic: fix skipping of check command

### PR DESCRIPTION
**Pull request by kai-tub**
## Description of changes

This commit fixes the (unnecessary) requirement to provide a `pruneOption` to ensure that the check command is run when a check option is set. This is useful for check-only configurations.

PS: This is my first time contributing to a service. Let me know if I should change anything!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pSub — APPROVED** [review]

File: nixos/modules/services/backup/restic.nix [hunk 1]
```diff
@@ -298,7 +298,9 @@ in
             doBackup = (backup.dynamicFilesFrom != null) || (backup.paths != null && backup.paths != []);
             pruneCmd = optionals (builtins.length backup.pruneOpts > 0) [
               (resticCmd + " forget --prune " + (concatStringsSep " " backup.pruneOpts))
-              (resticCmd + " check " + (concatStringsSep " " backup.checkOpts))
+            ];
+            checkCmd = optionals (builtins.length backup.checkOpts > 0) [
+                (resticCmd + " check " + (concatStringsSep " " backup.checkOpts))
             ];
```
Comment (MinerSebas) [inline nixos/modules/services/backup/restic.nix:304]: This check is annoying, as it is valid to call `restic check` without check specific options.
You should introduce a new Option to configure whether `check` is run. (For ease of use the default value of that option can then depend on `builtins.length backup.checkOpts > 0`)

**Review by MinerSebas — CHANGES_REQUESTED** [review]
The description for `checkOpts` is no longer accurate.

File: nixos/modules/services/backup/restic.nix (hunk 1 above)
Comment (kai-tub) [reply nixos/modules/services/backup/restic.nix]: :man_facepalming: I completely forgot the 'default' check version. I have added a `runCheck` option that defaults to `length backups.checkOpts > 0`.

**Comment by kai-tub** [comment]
Is there anything else I can address?

**Pull request merged**